### ISSUE TITLE
[sort-type-union-intersection-members] `void` should be in `nullish` group 

### ISSUE THREAD

**Issue by avaly** [typescript-eslint/typescript-eslint#2940]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

`void` is currently included in the `keyword` group, but it fits better with the `nullish` group.

```JSON
{
  "rules": {
    "@typescript-eslint/sort-type-union-intersection-members": [2, {
      "groupOrder": [
        "literal",
        "keyword",
        "named",
        "operator",
        "function",
        "import",
        "conditional",
        "object",
        "tuple",
        "intersection",
        "union",
        "nullish"
      ]
  }
}
```

```TS
type Foo = void | ObjectId | null | undefined;
```

**Expected Result**

```TS
type Foo = ObjectId | null | undefined | void;
```

**Actual Result**

```TS
type Foo = void | ObjectId | null | undefined;
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.13.0` |
| `@typescript-eslint/parser`        | `4.13.0` |
| `TypeScript`                       | `4.1.3` |
| `ESLint`                           | `7.13.0` |
| `node`                             | `14.15.0` |
